src/aiperf/endpoints/huggingface_generate.py

```diff
@@ -0,0 +1,132 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import json
+from typing import Any
+
+from aiperf.common.decorators import implements_protocol
+from aiperf.common.enums import EndpointType
+from aiperf.common.factories import EndpointFactory
+from aiperf.common.models import ParsedResponse
+from aiperf.common.models.metadata import EndpointMetadata
+from aiperf.common.models.record_models import RequestInfo
+from aiperf.common.protocols import EndpointProtocol, InferenceServerResponse
+from aiperf.endpoints.base_endpoint import BaseEndpoint
+
+
+@implements_protocol(EndpointProtocol)
+@EndpointFactory.register(EndpointType.HUGGINGFACE_GENERATE)
+class HuggingFaceGenerateEndpoint(BaseEndpoint):
+    """Hugging Face TGI (Text Generation Inference) endpoint.
+
+    Supports both non-streaming (/ or /generate) and streaming (/generate_stream)
+    endpoints automatically, based on the model endpoint's `streaming` flag.
+    """
+
+    @classmethod
+    def metadata(cls) -> EndpointMetadata:
+        """Return endpoint metadata for TGI."""
+        return EndpointMetadata(
+            endpoint_path="/generate",
+            streaming_path="/generate_stream",
+            supports_streaming=True,
+            produces_tokens=True,
+            tokenizes_input=True,
+            metrics_title="LLM Metrics",
+        )
+
+    def format_payload(self, request_info: RequestInfo) -> dict[str, Any]:
+        """Format payload for Hugging Face TGI request."""
+        if len(request_info.turns) != 1:
+            raise ValueError("TGI endpoint supports a single turn per request.")
+
+        turn = request_info.turns[0]
+        model_endpoint = request_info.model_endpoint
+
+        inputs = " ".join(
+            [content for text in turn.texts for content in text.contents if content]
+        )
+
+        payload: dict[str, Any] = {"inputs": inputs}
+
+        parameters: dict[str, Any] = {}
+        if turn.max_tokens is not None:
+            parameters["max_new_tokens"] = turn.max_tokens
+
+        if model_endpoint.endpoint.extra:
+            parameters.update(model_endpoint.endpoint.extra)
+
+        payload["parameters"] = parameters
```

Comment:
I think it is a little bit easier to read if you move the creation of the payload down to the bottom and create them inline together:
```python
payload: dict[str, Any] = {
    "inputs": inputs,
    "parameters": parameters
}
```